druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/URIExtractionNamespaceCacheFactory.java

```diff
@@ -155,11 +156,22 @@ public InputStream openStream() throws IOException {
                         }
                     };
                 }
-                final long lineCount = new MapPopulator<>(
+
+                final Map<String, List<String>> newCache = new ConcurrentHashMap<>();
+                final MapPopulator.PopulateResult populator = new MapPopulator<>(
                         extractionNamespace.getNamespaceParseSpec().getParser()).populate(source,
-                        cache).getLines();
-                log.info("Finished loading %d lines for namespace [%s]", lineCount, id);
-                return version;
+                        newCache);
+
+                if (newCache.size() == populator.getEntries()) {
+                    log.info("Loaded data to a new map successfully. Cleaning previous cache and adding data in the new map to cache ...");
+                    cache.keySet().retainAll(newCache.keySet());
```

Comment:
can you please make sure that `cache` is instance of ConcurrentHashMap ?

Reply:
sure, it is instance of `ConcurrentHashMap`. 

The `cache` in this method is coming from the following code, which call `getCacheMap()` which has abstract type `ConcurrentMap<String, U>` as its return type:
https://github.com/yahoo/maha/blob/master/druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/cache/MahaNamespaceExtractionCacheManager.java#L327

And `getCacheMap()` is override in the following part which initializes with `ConcurrentHashMap`
https://github.com/yahoo/maha/blob/master/druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/cache/OnHeapMahaNamespaceExtractionCacheManager.java#L69

Reply:
cool